armi/utils/hexagon.py

```diff
@@ -145,3 +145,50 @@ def numPositionsInRing(ring):
         rings is indexed to 1, i.e. the centermost position in the lattice is ``ring=1``.
     """
     return (ring - 1) * 6 if ring != 1 else 1
+
+
+def totalPositionUpToRing(ring: int) -> int:
```

Comment:
This could use some unit tests.

For instance, something like:

```python
def test_totalPositionUpToRing(self):
    self.assertEqual(totalPositionUpToRing(1), 1)
    self.assertEqual(totalPositionUpToRing(3), 19)

    for r in range(1, 10):
        tot = sum([numPositionsInRing(rr) for rr in range(1, r + 1)])
        self.assertEqual(totalPositionUpToRing(r), tot)
```

Just thinking aloud.

Reply:
Good call